### PR TITLE
[wasm2js] Run LLVM nontrapping fptoint lowering when running for emscripten

### PR DESCRIPTION
**Pull request by dschuff**
Lower away saturating fptoint operations when we know we are using emscripten.

### CODE REVIEW

**Comment by dschuff** [comment]
This is needed to pass some wasm2js tests in Emscripten. The alternative would be to implement full support for nontrapping fp operations in wasm2js, but I don't have a good sense for how much work that would be.

File: test/wasm2js/conversions-emscripten-modified.wast [hunk 1]
```diff
@@ -0,0 +1,564 @@
+(module
```
Comment (kripken) [inline test/wasm2js/conversions-emscripten-modified.wast:1]: There should be output files to check in with this, something like `FILENAME.2asm.js.*`.

But, for a test, maybe add just a minimal file with one fptoint operation? I don't think we need to add coverage for the entire set of instructions, which the pass has tests for. We mainly want to test that the pass runs. Unless there is a benefit I'm missing?

**Review by kripken — COMMENTED** [review]
lgtm % test

File: test/wasm2js/conversions-emscripten-modified.wast (hunk 1 above)
Comment (dschuff) [reply test/wasm2js/conversions-emscripten-modified.wast:1]: Ah yes, I forgot to check that output file in. but... ok why didn't the CI tests fail then? 🤔 
I'll switch to a minimal file.

Comment (kripken) [reply test/wasm2js/conversions-emscripten-modified.wast:1]: Hmm,

https://github.com/WebAssembly/binaryen/blob/cc97853b5f03e8c8441159e68ecb2262beee166f/scripts/test/wasm2js.py#L67-L68

It looks like we take the lack of an expected file to mean there is nothing there to test. That was needed for spec tests, as not all pass in wasm2js, but we shouldn't be doing that for normal tests... I'll investigate.

**Review by kripken — APPROVED** [review]

**Pull request merged**